cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/cna/CVRCnaDataReader.java

```diff
@@ -63,9 +62,8 @@ public class CVRCnaDataReader implements ItemStreamReader<CompositeCnaRecord>{
     private MultiKeyMap cnaMap = new MultiKeyMap();
     private Set<String> genes = new HashSet<>();
     private Set<String> samples = new HashSet<>();
-    private Set<String> newSamples = new HashSet<>();
 
-    private List<CompositeCnaRecord> cnaRecords = new ArrayList();
+    private final Deque<String> cnaRecords = new LinkedList<>();
```

Comment:
Anywhere I saw a `.remove(0);` on a list, I replaced the list with a `LinkedList` and called `.pollFirst()`.  Using a `LinkedList` instead of a `List` made a big difference.  `List`s are good if you want to be able to access any element in the list easily, but we only need to add to the end and then pull off the front.  A `LinkedList` is better for that.